cborg/src/Codec/CBOR/Read.hs

```diff
@@ -1036,13 +1067,27 @@ go_slow da bs !offset = do
       where
         !offset' = offset + fromIntegral (BS.length bs - BS.length bs')
 
+    SlowConsumeTokenByteArray bs' k len -> do
+      (bstr, bs'') <- getTokenVarLen len bs' offset'
+      let !str = BA.fromByteString bstr
+      lift (k str) >>= \daz -> go_slow daz bs'' (offset' + fromIntegral len)
+      where
+        !offset' = offset + fromIntegral (BS.length bs - BS.length bs')
+
     SlowConsumeTokenString bs' k len -> do
       (bstr, bs'') <- getTokenVarLen len bs' offset'
       let !str = T.decodeUtf8 bstr  -- TODO FIXME: utf8 validation
       lift (k str) >>= \daz -> go_slow daz bs'' (offset' + fromIntegral len)
       where
         !offset' = offset + fromIntegral (BS.length bs - BS.length bs')
 
+    SlowConsumeTokenUtf8ByteArray bs' k len -> do
+      (bstr, bs'') <- getTokenVarLen len bs' offset'
+      let !str = BA.fromByteString bstr  -- TODO FIXME: utf8 validation
```

Comment:
We should make up our mind, either we say that `ConsumeStringUTF8ByteArray` does validate the UTF8 or that it does not. Currently we document that it does not, so this TODO is wrong, or this TODO is right and we should do it and document that we do validate the UTF8.

Reply:
Alright, let's not validate in that case; dropping the TODO.